### PR TITLE
onesync: implement custom player scope radius natives

### PR DESCRIPTION
**Pull request by PichotM**
https://media.giphy.com/media/loLPei2rdrMqotZ4Jd/giphy.gif

### CODE REVIEW

File: code/components/citizen-server-impl/src/state/ServerGameState.cpp [hunk 1]
```diff
@@ -683,6 +685,13 @@ void ServerGameState::Tick(fx::ServerInstanceBase* instance)
 
 		auto slotId = client->GetSlotId();
 
+		float scopeRadius = serverScopeRadius;
+
+		if (auto clientScopeRadius = client->GetData("scopeRadius"); clientScopeRadius.has_value())
```
Comment (blattersturm) [inline code/components/citizen-server-impl/src/state/ServerGameState.cpp:690]: Please don't use the data bag for something that might get looked up a lot! Add a field to ClientSyncData (with an external accessor in ServerGameState.cpp?), perhaps.

File: code/components/citizen-server-impl/include/state/ServerGameState.h [hunk 2]
```diff
@@ -508,6 +508,16 @@ class ServerGameState : public fwRefCountable, public fx::IAttached<fx::ServerIn
 		m_entityLockdownMode = mode;
 	}
 
+	inline float GetEntityScopeRadius()
+	{
+		return m_entityScopeRadius;
+	}
+
+	inline void SetEntityScopeRadius(float scopeRadius)
```
Comment (blattersturm) [inline code/components/citizen-server-impl/include/state/ServerGameState.h:516]: Is this meant to be exposed by a native?

Comment (PichotM) [reply code/components/citizen-server-impl/include/state/ServerGameState.h:516]: if ever a 'global' radius can be set, yeah why not

File: code/components/citizen-server-impl/src/state/ServerGameState.cpp (hunk 1 above)
Comment (PichotM) [reply code/components/citizen-server-impl/src/state/ServerGameState.cpp:690]: Alright

**Comment by PichotM** [comment]
new pr soon

**Pull request closed without merging**